subsys/nrf_security/src/drivers/cracen/cracenpsa/src/common.c

```diff
@@ -570,7 +580,48 @@ static int cracen_prepare_ik_key(const uint8_t *user_data)
 		nrf_cracen_seedram_lock_enable_set(NRF_CRACEN, true);
 	}
 #endif
-	return sx_pk_ik_derive_keys(NULL);
+	struct sx_pk_config_ik cfg;
+	uint32_t device_secret[4] = {};
+
+	cfg.device_secret = device_secret;
+	cfg.device_secret_sz = ARRAY_SIZE(device_secret);
+
+	switch (user_data[1]) {
+#define SET_STR(x)                                                                                 \
+	{                                                                                          \
+		const char lstr_##x[((sizeof(x) + 3) / 4) * 4] = #x;                               \
+		cfg.key_bundle = (uint32_t *)lstr_##x;                                             \
+		cfg.key_bundle_sz = sizeof(lstr_##x) / sizeof(uint32_t);                           \
+	}
```

Comment:
Maybe add a short comment on what this does, especially as has all of these nice magic numbers in there.